sysdump/sysdump.go

```diff
@@ -1186,7 +1186,7 @@ func (c *Collector) submitCiliumBugtoolTasks(ctx context.Context, pods []*corev1
 			}()
 
 			// Run 'cilium-bugtool' in the pod.
-			command := append([]string{ciliumBugtoolCommand, "--archiveType=gz"}, c.Options.CiliumBugtoolFlags...)
+			command := append([]string{ciliumBugtoolCommand, "--archiveType=gz", "--exclude-object-files"}, c.Options.CiliumBugtoolFlags...)
```

Comment:
AFAICS, https://github.com/cilium/cilium/pull/22370 is only available in current Cilium `master` branch. However, `cilium sysdump` should also be usable with all stable releases still (currently v1.10.x, v1.11.x and v1.12.x). I think we'd need to guard the use of the `--exclude-object-files` flag on the Cilium version running in the pod.

Reply:
Oh, right :facepalm: It doesn't just ignore unknown flags.

Reply:
@tklauser @pchaigno That's a good point!
An approach could be to verify the version of the image in the loop, something like this:

```go
// get cilium agent version from container image
var version string
for _, container := range p.Spec.Containers {
	if container.Name == defaults.AgentContainerName {
                // e.g: quay.io/cilium/cilium:v1.12.2
		subs := strings.Split(container.Image, ":")
		version = subs[len(subs)-1]
	}
}

// pseudo-code here
flags := []string{"--archiveType=gz"}
if weHaveAVersion && semver.GTE(hardCodedNextVersionWithNewFlag) {
    flags = append(flags, "--exclude-object-files")
}

command := append([]string{ciliumBugtoolCommand}, flags, c.Options.CiliumBugtoolFlags...)
```

Would that make sense or is there a _more sophisticated_ way you already handle versions somewhere?

Reply:
That makes sense @tklauser, I updated the PR and left a comment Cc @pchaigno.